server/pulp/server/managers/repo/unit_association.py

```diff
@@ -139,7 +139,30 @@ def associate_all_by_ids(self, repo_id, unit_type_id, unit_id_list):
         return unique_count
 
     @staticmethod
-    def associate_from_repo(source_repo_id, dest_repo_id, criteria=None,
+    def _units_from_criteria(source_repo, criteria):
```

Comment:
In my opinion, this would be a nice thing to have on a custom QuerySet. I can point you at all the places you need to make changes, if you aren't already familiar with them.